jsign-crypto/src/main/java/net/jsign/jca/VenafiSigningService.java

```diff
@@ -0,0 +1,272 @@
+        }
+
+    }
+
+    @Override
+    public SigningServicePrivateKey getPrivateKey(String alias, char[] password) throws UnrecoverableKeyException {
+        try {
+            Certificate[] chain = getCertificateChain(alias);
+            String algorithm = chain[0].getPublicKey().getAlgorithm();
+            return new SigningServicePrivateKey(alias, algorithm, this);
+        } catch (KeyStoreException e) {
+            throw (UnrecoverableKeyException) new UnrecoverableKeyException().initCause(e);
+        }
+
+    }
+
+    /* This function encodes the ECDSA signature response from Venafi CodeSign Protect /vedhsm/api/sign REST API */
+    public byte[] encodeASN1Signature(byte[] sigBytes) throws IOException {
+
+        // Split the sigbytes into r and s components
+        BigInteger r = new BigInteger(1, Arrays.copyOfRange(sigBytes, 0, sigBytes.length / 2));
+        BigInteger s = new BigInteger(1, Arrays.copyOfRange(sigBytes, sigBytes.length / 2, sigBytes.length));
+
+        // Create an ASN1 sequence containing r and s
+        DLSequence components = new DLSequence(new ASN1Encodable[] {
+            new ASN1Integer(r),
+            new ASN1Integer(s)
+        });
+
+        // Marshal the components to ASN1 encoding
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        ASN1OutputStream asn1OutputStream = ASN1OutputStream.create(byteArrayOutputStream);
+        asn1OutputStream.writeObject(components);
+        asn1OutputStream.close();
+
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    @Override
+    public byte[] sign(SigningServicePrivateKey privateKey, String algorithm, byte[] data) throws GeneralSecurityException {
+        Integer clientMechanism = algorithmMapping.get(algorithm);
+        if (clientMechanism == null) {
+            throw new InvalidAlgorithmParameterException("Unsupported signing algorithm: " + algorithm);
+        }
+
+        try {
+            DigestAlgorithm digestAlgorithm = DigestAlgorithm.of(algorithm.substring(0, algorithm.toLowerCase().indexOf("with")));
+            data = digestAlgorithm.getMessageDigest().digest(data);
+            
+            byte[] arr_combined = addASN1Prefix(digestAlgorithm, data);
+
+            Map<String, Object> request = new HashMap<>();
+
+            Map<String, String> clientInfo = new HashMap<>();
+            clientInfo.put("ClientLibraryName", "jsign");
+
+            Map<String, String> processInfo = new HashMap<>();
+            clientInfo.put("Executable", "jsign");
+
+            request.put("ClientInfo", clientInfo);
+            request.put("ProcessInfo", processInfo); 
+            request.put("KeyId", KeyId);
+            request.put("Data", Base64.getEncoder().encodeToString(arr_combined));
+            request.put("ClientMechanism", clientMechanism);
```

Comment:
The `ClientMechanism` field seems to be optional, does it still work if removed?